.github/workflows/base.yml

```diff
@@ -508,6 +525,36 @@ jobs:
             thisRepo="${GITHUB_REPOSITORY}" \
             runId="${GITHUB_RUN_ID}"
 
+  native-tests-stats-upload:
+    name: Upload build stats to collector
+    if: ${{ always() && inputs.build-stats-tag != 'null' }}
+    needs:
+      - native-tests
+      - get-test-matrix
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.get-test-matrix.outputs.tests-matrix) }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: graalvm/mandrel
+          fetch-depth: 1
+          path: workflow-mandrel
+      - uses: actions/download-artifact@v1
+        with:
+          name: build-stats-${{matrix.category}}-${{ needs.get-test-matrix.outputs.artifacts-suffix }}
+          path: .
+      - name: Extract and import build stats
```

Comment:
Wouldn't it be simple to do the upload as part of the `native-tests` job (in place of the `Upload build JSON stats` step)?

Reply:
It would, but on the other hand I wanted to have a way to get the raw json files GraalVM produces for those builds too. That's useful sometimes for sanity checking and or when the upload fails.